### PR TITLE
Adds pulp-smash test for ansible remote

### PR DESCRIPTION
**Pull request by dkliban**

### CODE REVIEW

File: .travis/before_script.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+sudo mkdir -p /var/lib/pulp/tmp
+sudo mkdir /var/cache/pulp
+sudo mkdir /etc/pulp/
+sudo chown -R travis:travis /var/lib/pulp
+sudo chown travis:travis /var/cache/pulp
+
+if [ "$DB" = 'postgres' ]; then
+  sudo cp ../pulp/.travis/server.postgres.yaml /etc/pulp/server.yaml
+else
+  # docs job also requires server.yaml
```
Comment (daviddavis) [inline .travis/before_script.sh]: copy pasta. there is no docs job.

File: .travis/install.sh [hunk 2]
```diff
@@ -1,4 +1,35 @@
...
+pip install flake8 pytest
+
+cd .. && git clone https://github.com/pulp/pulp.git
+
+if [ -z $PULP_PR_NUMBER ]; then
+  pushd pulp && git checkout 3.0-dev && popd
+else
+  export PULP_SHA=$(curl https://api.github.com/repos/pulp/pulp/pulls/$PULP_PR_NUMBER | jq -r '.merge_commit_sha')
+  pushd pulp && git fetch origin +refs/pull/$PULP_PR_NUMBER/merge
+  git checkout $PULP_SHA && popd
```
Comment (daviddavis) [inline .travis/install.sh]: Mind breaking up these two lines onto multiple?

```
pushd pulp
git fetch origin +refs/pull/$PULP_PR_NUMBER/merge
git checkout $PULP_SHA
popd
```

File: .travis/install.sh [hunk 3]
```diff
@@ -1,4 +1,35 @@
...
+pushd pulp/plugin/ && pip install -e .  && popd
+
+if [ -z $PULP_SMASH_PR_NUMBER ]; then
+  pip install git+https://github.com/PulpQE/pulp-smash.git#egg=pulp-smash
+else
+  export PULP_SMASH_SHA=$(curl https://api.github.com/repos/PulpQE/pulp-smash/pulls/$PULP_SMASH_PR_NUMBER | jq -r '.merge_commit_sha')
+  git clone https://github.com/PulpQE/pulp-smash.git
+  pushd pulp-smash && git fetch origin +refs/pull/$PULP_SMASH_PR_NUMBER/merge
+  git checkout $PULP_SMASH_SHA
+  pip install -e . && popd
```
Comment (daviddavis) [inline .travis/install.sh]: Same here.

File: tests/pulp_smash/test_crud_remotes.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        """Delete an remote."""
+        self.client.delete(self.remote['_href'])
+        with self.assertRaises(HTTPError):
+            self.client.get(self.remote['_href'])
+
+
+def _gen_remote():
+    """Return a semi-random dict for use in creating an remote."""
+    return {
+        'download_policy': 'immediate',
```
Comment (asmacdo) [inline tests/pulp_smash/test_crud_remotes.py]: download_policy and sync_mode are no longer a part of the base remote.

**Comment by omaciel** [comment]
@daviddavis for the time being, my only suggestion is that you use the following layout for where your tests should reside:

`tests/functional/test_crud_remotes.py`

In other words, rename the `pulp_smash` folder to `functional`. This has the added bonus of allowing you to create a future `unit` folder for unit tests

**Comment by dkliban** [comment]
I'll update this PR based on the feedback.

**Pull request merged**